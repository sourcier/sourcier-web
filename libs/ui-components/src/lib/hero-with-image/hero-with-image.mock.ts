export const imageUrl = 'https://source.unsplash.com/random';

export const imageAlt = 'Random unsplash image';

export const heading = 'Hello there';

export const copy =
  'Provident cupiditate voluptatem et in. Quaerat fugiat ut assumenda excepturi exercitationem quasi. In deleniti eaque aut repudiandae et a id nisi.';
