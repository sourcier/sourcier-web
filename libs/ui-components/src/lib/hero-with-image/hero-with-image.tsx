import clsx from 'clsx';

export interface HeroWithImageProps {
  imageUrl: string;
  imageAlt: string;
  heading: string;
  copy: string;
  className?: string;
}

export function HeroWithImage({
  imageUrl,
  imageAlt,
  heading,
  copy,
  className,
}: HeroWithImageProps) {
  return (
    <div
      className={clsx(
        'min-h-[25vh] hero rounded-lg my-8 bg-gradient-to-t from-base-200 to-base-100 p-4 shadow-sm',
        className
      )}
    >
      <div className="flex-row-reverse hero-content">
        <img
          src={imageUrl}
          alt={imageAlt}
          className="max-w-sm rounded-lg shadow-2xl lg:max-w-xl"
        />
        <div>
          <h1 className="text-5xl font-bold">{heading}</h1>
          <p className="py-6">{copy}</p>
        </div>
      </div>
    </div>
  );
}

export default HeroWithImage;
