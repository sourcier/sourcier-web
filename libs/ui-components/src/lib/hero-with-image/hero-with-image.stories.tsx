import { Story, Meta } from '@storybook/react';

import { HeroWithImage, HeroWithImageProps } from './hero-with-image';
import { imageUrl, heading, copy, imageAlt } from './hero-with-image.mock';

export default {
  component: HeroWithImage,
  title: 'HeroWithImage',
} as Meta;

const Template: Story<HeroWithImageProps> = (args) => (
  <HeroWithImage {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  heading,
  copy,
  imageUrl,
  imageAlt,
};
