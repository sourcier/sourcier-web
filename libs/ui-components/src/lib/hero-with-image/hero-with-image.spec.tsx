import renderer from 'react-test-renderer';

import HeroWithImage from './hero-with-image';
import { imageUrl, imageAlt, heading, copy } from './hero-with-image.mock';

describe('HeroWithImage', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(
        <HeroWithImage
          copy={copy}
          heading={heading}
          imageUrl={imageUrl}
          imageAlt={imageAlt}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
