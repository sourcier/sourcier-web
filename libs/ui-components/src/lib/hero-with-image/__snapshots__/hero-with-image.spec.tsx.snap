// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroWithImage should render successfully 1`] = `
<div
  className="min-h-[25vh] hero rounded-lg my-8 bg-gradient-to-t from-base-200 to-base-100 p-4 shadow-sm"
>
  <div
    className="flex-row-reverse hero-content"
  >
    <img
      alt="Random unsplash image"
      className="max-w-sm rounded-lg shadow-2xl lg:max-w-xl"
      src="https://source.unsplash.com/random"
    />
    <div>
      <h1
        className="text-5xl font-bold"
      >
        Hello there
      </h1>
      <p
        className="py-6"
      >
        Provident cupiditate voluptatem et in. Quaerat fugiat ut assumenda excepturi exercitationem quasi. In deleniti eaque aut repudiandae et a id nisi.
      </p>
    </div>
  </div>
</div>
`;
